akka-javasdk/src/main/java/akka/javasdk/consumer/MessageContext.java

```diff
@@ -22,4 +21,21 @@ public interface MessageContext extends MetadataContext {
 
   /** Access to tracing for custom app specific tracing. */
   Tracing tracing();
+  /**
+   * Returns true if this message was originated on the same region that is being processed.
+   *
+   * <p>For messages coming from Akka entities:
+   *
+   * <ul>
+   *   <li>If the message is an event from an Event Sourced Entity, it returns true if the event was
+   *       originated in the region where this consumer is running. Otherwise, it returns false.
+   *   <li>If the message is a change update from a KeyValueEntity, it returns true if the update
+   *       was originated in the region where this consumer is running. Otherwise, it returns false.
```

Comment:
Already foreseeing KeyValueEntity replication. But in practice, as of today, change updates are only consumer on the original region anyway.